lxd/instance/drivers/driver_qemu.go

```diff
@@ -2185,7 +2185,16 @@ func (d *qemu) deviceAttachPath(deviceName string, configCopy map[string]string,
 		return fmt.Errorf("Failed to connect to QMP monitor: %w", err)
 	}
 
-	addr, err := net.ResolveUnixAddr("unix", virtiofsdSockPath)
+	virtiofsdSocketDir, err := os.Open(filepath.Dir(virtiofsdSockPath))
+	if err != nil {
+		return err
+	}
+
+	defer func() { _ = virtiofsdSocketDir.Close() }()
+
+	shortPath := fmt.Sprintf("/proc/self/fd/%d/%s", virtiofsdSocketDir.Fd(), filepath.Base(virtiofsdSockPath))
```

Comment:
Can you use  util.ShortenedSocketPath here instead?

Reply:
Like you did for swtpm?

Reply:
I can't because of what I explained to @roosterfish [here](https://github.com/canonical/lxd/pull/13320#discussion_r1617027761). It is a subtle difference in behavior, in short, in this case the file descriptor used is an LXD process file descriptor and `ShortenedSocketPath` uses another proccess's file descriptor via `AddFileDescriptor`. So if I used it here the value used on `%d` in "/proc/self/fd/%d/%s" would probably be wrong.